nautobot/core/tests/test_settings_schema.py

```diff
@@ -0,0 +1,223 @@
+            "LOGIN_URL",
+            "LOGIN_REDIRECT_URL",
+            "MEDIA_URL",
+            "MESSAGE_TAGS",
+            "MIDDLEWARE",
+            "PROMETHEUS_EXPORT_MIGRATIONS",
+            "REST_FRAMEWORK",
+            "REST_FRAMEWORK_ALLOWED_VERSIONS",
+            "REST_FRAMEWORK_VERSION",
+            "ROOT_URLCONF",
+            "SECURE_PROXY_SSL_HEADER",
+            "SHELL_PLUS_DONT_LOAD",
+            "SILKY_ANALYZE_QUERIES",
+            "SILKY_AUTHENTICATION",
+            "SILKY_AUTHORISATION",
+            "SILKY_INTERCEPT_FUNC",
+            "SILKY_PERMISSIONS",
+            "SILKY_PYTHON_PROFILER",
+            "SILKY_PYTHON_PROFILER_BINARY",
+            "SILKY_PYTHON_PROFILER_EXTENDED_FILE_NAME",
+            "SOCIAL_AUTH_POSTGRES_JSONFIELD",
+            "SPECTACULAR_SETTINGS",
+            "STATIC_URL",
+            "STATICFILES_DIRS",
+            "TEMPLATES",
+            "TESTING",
+            "TEST_RUNNER",
+            "USE_I18N",
+            "USE_TZ",
+            "USE_X_FORWARDED_HOST",
+            "VERSION",
+            "VERSION_MAJOR",
+            "VERSION_MINOR",
+            "WEBSERVER_WARMUP",
+            "WSGI_APPLICATION",
+            "X_FRAME_OPTIONS",
+        ]
+        keys = sorted(nautobot_settings.__dict__.keys())
+
+        # Make sure the above list is up to date and doesn't contain cruft
+        for key in FROZEN_SETTINGS:
+            with self.subTest(f"Checking for FROZEN_SETTINGS entry {key} in nautobot.core.settings"):
+                self.assertIn(key, keys)
+
+        keys = [key for key in keys if key == key.upper() and not key.startswith("_") and key not in FROZEN_SETTINGS]
+
+        for key in keys:
+            with self.subTest(f"Checking for settings attribute {key} in the settings schema"):
+                self.assertIn(key, self.schema_data["properties"])
+            if key not in self.schema_data["properties"]:
+                continue
+
+            with self.subTest(f"Checking default value for settings attribute {key} against the schema"):
+                if self.schema_data["properties"][key].get("$ref", None) in [
+                    "#/definitions/absolute_path",
+                    "#/definitions/callable",
+                    "#/definitions/relative_path",
+                ]:
+                    # Functions like UI_RACK_VIEW_TRUNCATE_FUNCTION don't have a default expressable in JSON Schema,
+                    # and path-like defaults probably depend on the user context
+                    continue
+                elif self.schema_data["properties"][key].get("default_literal", None):
+                    # Probably a complicated setting that isn't easily documentable in JSON schema
+                    continue
+                else:
+                    # Is the default in the settings correctly documented?
+                    self.assertEqual(getattr(nautobot_settings, key), self.schema_data["properties"][key]["default"])
```

Comment:
Not blocking for this PR but I think it'd be great to consider as a follow-on item(s):

1. Add a test that if `environment_variable` key is defined, that mocking that environment variable and re-importing the settings correctly updates the settings?
2. Add a test that asserts that every variable in the schema has an `environment_variable` or variables appropriately defined? (Enforcing #1150 basically)

Reply:
Super rough draft of the test [here](https://github.com/nautobot/nautobot/pull/5363/commits/3565104cf9b62cf6692ae79e5d388bfe9781cf3e#diff-69f108c7d0d5b8ea13289dfab32fb7985b0701daadc743ce66c8ec1387f534afR233-R290)